### PR TITLE
ci: generate docker image in nodejs-release

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Generate docker image in `nodejs-release`:
- add needed envs to push Docker image to DockerHub

If someone want to release docker image have to install `@semantic-release-plus/docker` package in the project and add needed config to the `semantic-release` config. Also need to define Docker Image's build script in the `release` script. Example PR: https://github.com/asyncapi/server-api/pull/24

**Related issue(s)**
Part of https://github.com/asyncapi/server-api/pull/24

### CODE REVIEW

File: .github/workflows/if-nodejs-release.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
 
   release:
     needs: test
-    name: Publish to NPM and GitHub
+    name: Publish to NPM/DockerHub and GitHub
```
Comment (smoya) [inline .github/workflows/if-nodejs-release.yml:50]: Just small suggestion that makes sense to me (but can be obviously discarded)
```suggestion
    name: Publish to any of NPM, Github, and Docker Hub
```

File: .github/workflows/if-nodejs-release.yml [hunk 2]
```diff
@@ -68,11 +68,16 @@ jobs:
         name: Install dependencies
         run: npm install
       - if: steps.packagejson.outputs.exists == 'true'
-        name: Release to NPM and GitHub
+        name: Build Docker Image (if script exists)
```
Comment (smoya) [inline .github/workflows/if-nodejs-release.yml:71]: I would just delegate this to the `npm run release` script.

Comment (magicmatatjahu) [reply .github/workflows/if-nodejs-release.yml]: Good idea! I will change it :)

**Review by smoya — DISMISSED** [review]
LGTM 👍

File: .github/workflows/if-nodejs-release.yml [hunk 3]
```diff
@@ -68,7 +68,7 @@ jobs:
         name: Install dependencies
         run: npm install
       - if: steps.packagejson.outputs.exists == 'true'
-        name: Publish to any of NPM, Github, and Docker Hub
+        name: Release to any of NPM, Github, and Docker Hub
```
Comment (smoya) [inline .github/workflows/if-nodejs-release.yml:71]: I would keep `Publish` word here for consistency.

**Comment by magicmatatjahu** [comment]
@derberg You should be interested in this solution :) Of course it is not obligatory, maybe you have a better idea.

**Review by derberg — APPROVED** [review]
@magicmatatjahu definitely something we can add, release pipeline is still super flexible! especially in cases like generator, where we build docker in another workflow to build it from npm release and not sources. Nicely done!

**Comment by derberg** [comment]
let's merge it later in the day or I can if you want tomorrow morning. These updates always occupy CI a bit during the day

**Pull request merged**